src/org/zaproxy/zap/extension/pscanrulesAlpha/PiiScanner.java

```diff
@@ -0,0 +1,166 @@
+        CreditCardPatterns.put("DinersClub", DinersClub);
+        CreditCardPatterns.put("Discover", Discover);
+        CreditCardPatterns.put("Jcb", Jcb);
+        CreditCardPatterns.put("Maestro", Maestro);
+        CreditCardPatterns.put("Mastercard", Mastercard);
+        CreditCardPatterns.put("Visa", Visa);
+    }
+
+    public PiiScanner() {
+        //
+    }
+
+    @Override
+    public void setParent(PassiveScanThread parent) {
+        this.parent = parent;
+    }
+
+    @Override
+    public void scanHttpRequestSend(HttpMessage msg, int id) {
+
+    }
+
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        String responseBody = msg.getResponseBody().toString();
+        List<String> candidates = getAllNumberSequences(responseBody);
+        for(String candidate : candidates) {
+            for (String cardType : CreditCardPatterns.keySet()) {
+                Matcher matcher = CreditCardPatterns.get(cardType).matcher(candidate);
+                while (matcher.find()) {
+                    String evidence = matcher.group();
+                    if(validateLuhnChecksum(evidence)) {
+                        raiseAlert(msg, id, evidence, cardType);
+                    }
+                }
+            }
+        }
+    }
+
+    private boolean validateLuhnChecksum(String evidence) {
+        int sum = 0;
+        int parity = evidence.length() % 2;
+        for (int index = 0; index < evidence.length(); index++) {
+            int digit = Integer.parseInt(evidence.substring(index, index + 1));
+            if ((index % 2) == parity) {
+                digit *= 2;
+                if (digit > 9) {
+                    digit -= 9;
+                }
+            }
+            sum += digit;
+        }
+        return (sum % 10) == 0;
+    }
+
+    private void raiseAlert(HttpMessage msg, int id, String evidence, String cardType) {
+        Alert alert = new Alert(getPluginId(), Alert.RISK_HIGH, Alert.CONFIDENCE_HIGH,  getName());
+        alert.setDetail(
+                Constant.messages.getString(MESSAGE_PREFIX + "desc"),
+                msg.getRequestHeader().getURI().toString(),
+                "", //parameter
+                "",                     //attack
+                String.format("Credit Card Type detected: %s", cardType),
+                "",            //solution
+                "",
+                evidence,     //evidence, if any
+                359, //CWE-359: Exposure of Private Information ('Privacy Violation')
+                13,  //WASC-13: Information Leakage
+                msg);
+
+        parent.raiseAlert(id, alert);
+    }
+
+    private List<String> getAllNumberSequences(String inputString) {
+        return getAllNumberSequences(inputString, 3);
+    }
+
+    private List<String> getAllNumberSequences(String inputString, int minSequence) {
+        String regexString = String.format("((?:\\d{%d,}[\\s]*)+)", minSequence);
```

Comment:
The surrounding parentheses in the regular expression can be removed (the whole match can be obtained without them).